mountpoint-s3-client/src/object_client.rs

```diff
@@ -707,6 +711,28 @@ impl Checksum {
             checksum_sha256: None,
         }
     }
+
+    /// Provide [ChecksumAlgorithm] for the [Checksum], if set and recognized.
+    ///
+    /// This method assumes that at most one checksum will be set and will return the first matched.
+    /// This method cannot account for any new checksum algorithms that may be introduced in the future,
+    /// returning [None] when no algorithm is recognized.
```

Comment:
The last sentence does not seem relevant: it's `Checksum` itself that does not account for new algorithms, rather than this method in particular.

Reply:
Fair enough, will remove. We already have "if set and recognized" in the first line.